### DISCUSSION TITLE
Writing data values with a source timestamp from a client

### DISCUSSION THREAD

**Discussion by andrewwkoerner** [HMIProject/open62541#187] [General]
I'm hoping to write DataValues to variable nodes with source timestamp that comes from the client.

I reviewed the examples and library code, but I don't see how it's possible to update the source timestamp as a client when using the `write_value()`.

I don't see a function like `with_status_code()` to allow setting the source timestamp https://github.com/HMIProject/open62541/blob/main/src/ua/data_types/data_value.rs#L20

Is there a way to do this? Am I missing something.

Appreciate your help!

**Comment by sgoll** [comment]
You are right, these methods are missing at the moment. I have now added them in #188. Feel free to comment on that pull request if you have any remarks regarding the upcoming implementation.

**Comment by andrewwkoerner** [comment]
Appreciate it, thank you!
